### PR TITLE
Initialize arrays to zero before using in gaussian_sfcanl

### LINKED ISSUE #71: Initialize gaussian_data in gaussian_sfcanl
### What is wrong?

In the gaussian_sfcanl utility, the variable `gaussian_data` is used before it is initialized to zero ([Line 273](https://github.com/NOAA-EMC/gfs-utils/blob/6df304f6f09006593643a5a1a890e1f940132cb5/src/gaussian_sfcanl.fd/gaussian_sfcanl.f90#L273)):
```
  gaussian_data%vfrac(row(i))  = gaussian_data%vfrac(row(i)) + s(i)*tile_data%vfrac(col(i))
```
Sometimes the resulting values are unreasonable (for example 1E+200).

### What should have happened?

The `gaussian_data` variable should be initialized to zero after allocation.

### What machines are impacted?

All or N/A

### Steps to reproduce

- Clone and build the global-workflow
- Run the WCDA CI test at C48mx500 resolution
- Examine the gaussian sfcanl nc file and ncdump variable `veg`

### Additional information

More details are in [global-workflow issue #2669](https://github.com/NOAA-EMC/global-workflow/issues/2669).

### Do you have a proposed solution?

The `gaussian_data` variable should be initialized to zero after allocation.

### PR DESCRIPTION
**Pull request by GeorgeGayno-NOAA**
# Description
The gaussian_data array is initialized to zero after allocation. Not doing so can result in garbage values being used.

Resolves #71 
Refs https://github.com/NOAA-EMC/global-workflow/issues/2669

# Type of change
- Bug fix.

# Change characteristics
- Is this a breaking change (a change in existing functionality)? NO
- Does this change require a documentation update? NO

# How has this been tested?
Tests documented here (https://github.com/NOAA-EMC/gfs-utils/issues/71#issuecomment-2220570478) and here (https://github.com/NOAA-EMC/gfs-utils/issues/71#issuecomment-2220887710)

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [ ] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by GeorgeGayno-NOAA** [comment]
Are there official (i.e. regression tests) that need to run? How do I run them?

**Comment by CatherineThomas-NOAA** [comment]
@GeorgeGayno-NOAA: There's CI testing set up in the workflow that will be run when a PR to update the gfs_utils hash is submitted.  For now, let's just run our own tests.  I'll check out your hash and run a few cycles at C48 and C384.

**Comment by RussTreadon-NOAA** [comment]
Any update on this PR?  gdasatmanlprod still fails when running g-w C48mx500_3DVarAOWCDA CI.

**Comment by CatherineThomas-NOAA** [comment]
I ran 4 short S2S cycling tests on Hera: 2 at C48 and 2 at C384.  For each resolution, one pointed to gfs-utils from develop, the other pointed to this PR.  

The gdasatmanlprod step did not fail for any case, but the C48 develop case contained `-Infinityf` values in the gaussian sfcanl nc file.  The C48 PR case had reasonable values throughout.

Thanks @GeorgeGayno-NOAA for taking care of this.

**Review by CatherineThomas-NOAA — APPROVED** [review]

**Comment by CatherineThomas-NOAA** [comment]
@GeorgeGayno-NOAA: Do we need another reviewer before this can move forward?

**Comment by GeorgeGayno-NOAA** [comment]
> @GeorgeGayno-NOAA: Do we need another reviewer before this can move forward?

It looks like this PR can't be merged until it is reviewed by someone with write access. I don't know who they are.

**Comment by CatherineThomas-NOAA** [comment]
@aerorahul: Do you know who needs to review/merge PRs for this repo?  Is it the g-w team?

**Comment by aerorahul** [comment]
@CatherineThomas-NOAA 
I don't watch this repo.  Someone needs to tag me or let me know. 
I will merge this PR now based on your review.

**Review by aerorahul — APPROVED** [review]
looks good.

**Comment by CatherineThomas-NOAA** [comment]
Thanks @aerorahul!

**Pull request merged**
